tools/logger/convert.c

```diff
@@ -1125,5 +1131,8 @@ int convert(struct convert_config *config)
 	ret = logger_read();
 out:
 	free(config->uids_dict);
+clean_up:
+	config->logs_header = NULL;
+	global_config = NULL;
```

Comment:
Technically the second one doesn't point to the stack (so it's technically unneeded), but I'd say setting it to null is possibly fine here so I don't actually have an issue with this.

Reply:
global_config is not owned by this function so this looks wrong and possibly buggy. Did you test this? sof-logger is not used by Intel on this branch.

Reply:
technically inside this function we don't know where the `config` parameter comes from, but in fact it is on stack in the caller's function. A better fix would be to get rid of those global variables and to pass `config` as a parameter to all functions.

Reply:
@marc-hb this function is called only in one place. And after it completes `main()` completes too.